test/cpp/api/parallel.cpp

```diff
@@ -0,0 +1,179 @@
+#include <catch.hpp>
+
+#include <torch/csrc/autograd/functions/comm.h>
+#include <torch/nn/module.h>
+#include <torch/nn/modules/linear.h>
+#include <torch/nn/parallel/data_parallel.h>
+#include <torch/nn/pimpl.h>
+#include <torch/tensor.h>
+
+#include <iostream>
+#include <memory>
+#include <utility>
+#include <vector>
+
+using namespace torch::autograd;
+using namespace torch::nn;
+
+TEST_CASE("Parallel/DifferentiableScatter", "[cuda]") {
+  Scatter scatter(
+      {torch::Device(torch::kCUDA, 0), torch::Device(torch::kCUDA, 1)});
+
+  auto input = torch::ones(10, torch::requires_grad(true));
+  auto output = scatter.apply({input});
+
```

Comment:
Would be good to check that content of outputs is ok (move to a single device, use `cat`)